src/java/org/broadinstitute/dropseqrna/utils/SplitBamByCell.java

```diff
@@ -70,9 +73,12 @@
     @Argument(doc="The tag to examine in order to partition reads.")
     public String SPLIT_TAG="XC";
 
-    @Argument(doc="Number of output files to create")
+    @Argument(optional=true, doc="Number of output files to create")
```

Comment:
Instead of making NUM_OUTPUTS and TARGET_BAM_SIZE optional, you can use mutex flag in @Argument.  Then you don't need the code to check that one and only one is set.  Search our code for 'mutex' and you'll see how to use this.